onnxruntime/test/providers/cpu/tensor/unsqueeze_op_test.cc

```diff
@@ -63,5 +63,17 @@ TEST(TensorOpTest, Unsqueeze_OutOfRange) {
   test.Run(OpTester::ExpectResult::kExpectFailure, "Mismatch between number of source and target dimensions.");
 }
 
+// Disable this test given ONNX spec shape inference has bug.
+/*
```

Comment:
This is disabled because of there is a ONNX spec bug about valid Axis range, spec is saying about [-r, r-1], but even today r is a valid axis. PyTorch unsqueeze has different behavior than our current implementation, even without neg axis. We need to figure out the right behavior. I will send separate email to discuss this.

Reply:
The spec for `Unsqueeze` needs to be fixed. I have a PR to do so - https://github.com/onnx/onnx/pull/2347.

Once that fix is check-ed in, this test can be enabled again.

Reply:
If you wait for this change - https://github.com/microsoft/onnxruntime/pull/1907, you should be able to enable this test again. The newer ONNX commits should have fix for this.

Reply:
Merged from master with change, and re-enabled this test.